File: DDA/bubblesort.py
import time

unordered_list = [7,10,4,2,8,6]

# Custom bubble sorting algorithm
start_time = time.time()
swapped = True
while swapped:
    swapped = False
    for i in range(len(unordered_list)-1):
        if unordered_list[i] > unordered_list[i + 1]:
            swapped = True
            unordered_list[i], unordered_list[i + 1] = unordered_list[i + 1], unordered_list[i]
end_time = time.time()
custom_sort_time = end_time - start_time

print(f"Time taken by custom sort: {custom_sort_time} seconds")
print(unordered_list)

File: DDA/bubblesort.js
function bubbleSort(unorder_list) {
    let swapped = true;
    while (swapped) {
        swapped = false;
        for (let i = 0; i < unorder_list.length - 1; i++) {
            if (unorder_list[i] > unorder_list[i + 1]) {
                swapped = true;
                [unorder_list[i], unorder_list[i + 1]] = [unorder_list[i + 1], unorder_list[i]];
            }
        }
    }
    return unorder_list;
}

let unorder_list = [89, 4, 19, 2, 5, 3, 7, 11, 8, 10, 9, 6, 1, 12, 13, 14, 15, 16, 17, 18, 20];
console.log(bubbleSort(unorder_list));

